powerscale/provider/synciq_global_settings_resource.go

```diff
@@ -0,0 +1,578 @@
+/*
+Copyright (c) 2024 Dell Inc., or its subsidiaries. All Rights Reserved.
+
+Licensed under the Mozilla Public License Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://mozilla.org/MPL/2.0/
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package provider
+
+import (
+	"context"
+	powerscale "dell/powerscale-go-client"
+	"fmt"
+	"terraform-provider-powerscale/client"
+	"terraform-provider-powerscale/powerscale/constants"
+	"terraform-provider-powerscale/powerscale/helper"
+	"terraform-provider-powerscale/powerscale/models"
+
+	"github.com/hashicorp/terraform-plugin-framework-validators/stringvalidator"
+	"github.com/hashicorp/terraform-plugin-framework/attr"
+	"github.com/hashicorp/terraform-plugin-framework/schema/validator"
+	"github.com/hashicorp/terraform-plugin-framework/types"
+	"github.com/hashicorp/terraform-plugin-framework/types/basetypes"
+
+	"github.com/hashicorp/terraform-plugin-framework/diag"
+	"github.com/hashicorp/terraform-plugin-framework/resource"
+	"github.com/hashicorp/terraform-plugin-framework/resource/schema"
+	"github.com/hashicorp/terraform-plugin-log/tflog"
+)
+
+// Ensure provider defined types fully satisfy framework interfaces.
+var (
+	_ resource.Resource              = &SyncIQGlobalSettingsResource{}
+	_ resource.ResourceWithConfigure = &SyncIQGlobalSettingsResource{}
+	//_ resource.ResourceWithImportState = &SyncIQGlobalSettingsResource{}
+)
+
+// NewSyncIQGlobalSettingsResource creates a new resource.
+func NewSyncIQGlobalSettingsResource() resource.Resource {
+	return &SyncIQGlobalSettingsResource{}
+}
+
+// SyncIQGlobalSettingsResource defines the resource implementation.
+type SyncIQGlobalSettingsResource struct {
+	client *client.Client
+}
+
+// Metadata describes the resource arguments.
+func (r *SyncIQGlobalSettingsResource) Metadata(ctx context.Context, req resource.MetadataRequest, resp *resource.MetadataResponse) {
+	resp.TypeName = req.ProviderTypeName + "_synciq_global_settings"
+}
+
+// Schema describes the resource arguments.
+func (r *SyncIQGlobalSettingsResource) Schema(ctx context.Context, req resource.SchemaRequest, resp *resource.SchemaResponse) {
```

Comment:
What is the behaviour if nothing is specified?

Reply:
added the changes for that

Reply:
![image](https://github.com/user-attachments/assets/8cccc6ab-3400-41c1-87f7-84fec3948df6)